### PR TITLE
promotion's feature ad

### PR DESCRIPTION
**Pull request by arijit1111**
brand's promotion featured advertisement all over website by paying to the bazar

### CODE REVIEW

File: Modules/Promotion/Http/Services/PromotionService.php [hunk 1]
```diff
@@ -182,7 +232,10 @@ public function delete(string $promotionKey): array
             ];
         }
 
-        $promotion->delete();
+        $promotion->status = Promotion::STATUS_DEACTIVATED;
```
Comment (smaqeelkazmi) [inline Modules/Promotion/Http/Services/PromotionService.php:235]: @arijit1111 at this line: `$promotion = Promotion::where('promotion_key', $promotionKey)->first();` we should be validating if the promotion really belongs to the active user.

File: Modules/Promotion/Http/Services/PromotionService.php [hunk 2]
```diff
@@ -192,4 +245,17 @@ public function delete(string $promotionKey): array
...
     }
 
+    /**
+     * Get a listing of the promotion featured ads
+     *
+     * @return array
+     */
+    public function getFeatures(): array
+    {
+        $features = Feature::get();
```
Comment (smaqeelkazmi) [inline Modules/Promotion/Http/Services/PromotionService.php:255]: @arijit1111 Why we are getting the all `Feature`? How `Feature` are actually created?

**Review by smaqeelkazmi — CHANGES_REQUESTED** [review]
Please look into the queries @arijit1111

File: Modules/Promotion/Http/Services/PromotionService.php (hunk 1 above)
Comment (arijit1111) [reply Modules/Promotion/Http/Services/PromotionService.php:235]: applied in controller, please have a look

File: Modules/Promotion/Http/Services/PromotionService.php (hunk 2 above)
Comment (arijit1111) [reply Modules/Promotion/Http/Services/PromotionService.php:255]: Feature are Featured ad for promotion, table created by migration but data inserted manually

File: Modules/Promotion/Http/Services/PromotionService.php (hunk 1 above)
Comment (smaqeelkazmi) [reply Modules/Promotion/Http/Services/PromotionService.php:235]: @arijit1111 I still cannot see where you have applied.

File: Modules/Promotion/Http/Services/PromotionService.php (hunk 2 above)
Comment (smaqeelkazmi) [reply Modules/Promotion/Http/Services/PromotionService.php:255]: @arijit1111 So the questions is, why we are getting all of them instead for only the active user?

File: Modules/Promotion/Http/Services/PromotionService.php (hunk 1 above)
Comment (smaqeelkazmi) [reply Modules/Promotion/Http/Services/PromotionService.php:235]: @arijit1111 Thanks, I can see as its been validated through controller

File: Modules/Campaign/Http/Services/CampaignService.php [hunk 3]
```diff
@@ -1,152 +1,304 @@
...

-        ];

-    }

-

-}

+<?php
```
Comment (smaqeelkazmi) [inline Modules/Campaign/Http/Services/CampaignService.php:1]: @arijit1111 There is no usage of CampaignService, can you please avoid to include any not-required file in PR? You may need to delete it.

File: Modules/Campaign/Http/Services/CampaignService.php [hunk 4]
```diff
@@ -1,152 +1,304 @@
...
+
+
+
+        $allCampaignsCount = Campaign::where('user_id', $requestData->user_id)->count();
+
+        $draftCampaignsCount = Campaign::where('user_id', $requestData->user_id)->where('status', 'draft')->count();
+
+        $scheduledCampaignsCount = Campaign::where('user_id', $requestData->user_id)->where('status', 'schedule')->count();
+
+        $completedCampaignsCount = Campaign::where('user_id', $requestData->user_id)->where('status', 'completed')->count();
```
Comment (sherazbutt93) [inline Modules/Campaign/Http/Services/CampaignService.php]: @arijit1111 why are we querying 4 time for just the counts? Can't we query at once and get the counts based on status?
Like this:
$campaignCounts = Campaign::selectRaw('status, count(*) as count')
    ->where('user_id', $requestData->user_id)
    ->groupBy('status')
    ->get()
    ->pluck('count', 'status');

$allCampaignsCount = $campaignCounts->sum();
$draftCampaignsCount = $campaignCounts->get('draft', 0);
$scheduledCampaignsCount = $campaignCounts->get('schedule', 0);
$completedCampaignsCount = $campaignCounts->get('completed', 0);

You can test the query, but we should send the request to DB at once not four times.

File: Modules/Campaign/Http/Services/CampaignService.php [hunk 5]
```diff
@@ -1,152 +1,304 @@
...
+
+            'msg' => 'Campaign successfully deleted',
+
+            'data' => ""
+
+        ];
+
+    }
+
+
```
Comment (sherazbutt93) [inline Modules/Campaign/Http/Services/CampaignService.php]: @arijit1111 please remove all the empty spaces in the Campaign Service, it not proper aligned.

**Review by sherazbutt93 — COMMENTED** [review]
@arijit1111 please find the recent comments added by me and fix those so i can merge this PR.

**Comment by properweb** [comment]
Please have a look on the newest push, above comments are fixed

**Review by sherazbutt93 — APPROVED** [review]

**Pull request merged**